VERSIONING.md

```diff
@@ -0,0 +1,59 @@
+Rocket Chip Versioning Strategy ** All Strawman to be discussed**
+===============================
+
+Version Naming
+--------------
+
+* Rocket Chip uses a Calendar Versioning release strategy (https://calver.org/).
+* The format is YYYY.0M.0D.MINOR
+* Example:
+  * 2020.04.04.0 is the release which occurred on April 4, 2020.
+  * 2020.04.04.1 is a release that could occur at any time, but introduces only MINOR changes over the 2020.04.04.0 release
+  * 2020.06.01.0 is a release that occurs on June 1, 2020, with MAJOR (API breaking) changes over the 2020.04.04.X release.
+* MINOR changes include:
+  * API Additions
+  * Bug Fixes (which can result in changes in the emitted hardware considered bugs)
+* MINOR changes do NOT include:
+  * Changes in top-level I/Os for < level of hierarchy >
+  * Backwards incompatible changes in the API (see below)
+
+API
+---
+
+The Rocket-Chip API is: 
+  * All public Scala methods and classes
+  * All public Scala methods and classes of included Scala libraries (hardfloat, CDE, etc)
+  * The command line used to invoke the generator
+  * All Verilog interfaces (I/Os and parameterization) for verilog files
+  * C code...?
+  * The version(s) of Chisel, FIRRTL, rocket-tools pointed to by rocket-chip.
+  * Any code that generates IO(s) of any Module
+  * The cmd line interface to scripts like vlsi-mem-gen
+  * `public val`s within `LazyModule`s which translate to diplomatic nodes (?? -- this could go either way)
+
+The following is NOT part of the API:
+  * `public val`s within `Module`s which translate to Wires, Regs, module instances
+    * (We note that this may cause issues for people using Cross-Module references)
```

Comment:
I'm thinking i should remove this relaxation so that we can use more automated tooling to find changes

Reply:
What do you mean here?

Reply:
I *think* i am agreeing with your larger point below... people use the Scala API to do cross-module references, so we should NOT allow changes to `public vals` within Modules. This also lets us use normal binary compatability testing frameworks to detect these sort of changes.

None of this helps the person doing a verilog CMR, where things like annotations applied, .suggestName(s) applied, configuration settings might change, but this seems more straightforward to implement and audit to move this bullet up to the "part of the API" section above.